### PR TITLE
Feat/Community-Ribbon Component

### PR DESCRIPTION
**Pull request by nicmak**
## Related issues

#48 new: Promo/flag/ribbon/special marker

## Description

New Component for allowing users to see an item on sale or a special feature of an item

## Checklist before submitting pull request
- [x] New code is unit tested
- [x] make sure visual and accessibility tests pass
- [x] make sure code builds

 ## Output in npm prep r

tds-community-e2e | OK. 4 assertions passed. (2.75s)
tds-community-e2e | Running:  @tds/community-ribbon
tds-community-e2e |
tds-community-e2e | ✔ Element <#ribbon> was visible after 34 milliseconds.
tds-community-e2e | ✔ #rsg-root passes accessibility scan  - 212 ms.
tds-community-e2e | ✔ Passed [ok]: 11 aXe Tests Passed
tds-community-e2e | ✔ Screenshots Matched for chrome_headless.png with a tolerance of 0%, actual was 0%  - 196 ms.

### CODE REVIEW

File: packages/Ribbon/Ribbon.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```jsx
+<Ribbon ribbonCopy="Telus Ribbon" />
```
[the remark this answers is not in the document]
Comment (nicmak) [reply packages/Ribbon/Ribbon.md]: added

File: packages/Ribbon/Ribbon.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+```
+
+- `Ribbon` is a fixed width element and will take a string as a prop to display its content.
```
Comment (sketchidea) [inline packages/Ribbon/Ribbon.md]: This is not accurate, the component is not fixed width. The width adjusts to the content.

File: packages/Ribbon/Ribbon.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import RibbonWrapper from './styles'
+
+/** Ribbon component for focusing attention onto a sale or special feature.
+ * @version ./package.json
+ */
+const Ribbon = ({ ribbonCopy, ...rest }) => {
+  return (
+    <React.Fragment>
+      {<RibbonWrapper {...safeRest(rest)}>{ribbonCopy}</RibbonWrapper>}
```
Comment (charliehay) [inline packages/Ribbon/Ribbon.jsx]: No need to wrap this component in the curly braces.

**Comment by charliehay** [comment]
Still a few comments that need revision, please take a look when you have a chance and we can merge this early next week.

File: packages/Ribbon/Ribbon.jsx (hunk 3 above)
Comment (nicmak) [reply packages/Ribbon/Ribbon.jsx]: okay removed

File: packages/Ribbon/__tests__/Ribbon.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  ) => shallow(<Ribbon {...props} />)
+
+  it('matches snapshot with ribbonCopy prop', () => {
+    const ribbon = doShallow()
+    expect(ribbon).toMatchSnapshot()
+  })
+
+  it('matches snapshot with empty ribbonCopy prop', () => {
+    const ribbon = doShallow()
+    ribbon.setProps({ ribbonCopy: '' })
```
[the remark this answers is not in the document]
Comment (nicmak) [reply packages/Ribbon/__tests__/Ribbon.spec.jsx]:
```  it('matches snapshot with ribbonCopy prop', () => {
    const ribbon = doShallow({ ribbonCopy: 'TDS Ribbon' })
    expect(ribbon).toMatchSnapshot()
  })

   it('matches snapshot with empty ribbonCopy prop', () => {
    const ribbon = doShallow({ ribbonCopy: '' })
    expect(ribbon).toMatchSnapshot()
  })```

**Pull request merged**